server/tserver/src/main/java/org/apache/accumulo/tserver/FileManager.java

```diff
@@ -497,16 +491,16 @@ public void setInterruptFlag(AtomicBoolean flag) {
             + " files.size()=" + files.size() + " maxOpen=" + maxOpen + " tablet = " + tablet);
       }
 
-      List<FileSKVIterator> newlyReservedReaders = reserveReaders(tablet, files, continueOnFailure);
+      Map<FileSKVIterator,String> newlyReservedReaders = reserveReaders(tablet, files, continueOnFailure);
 
-      tabletReservedReaders.addAll(newlyReservedReaders);
+      tabletReservedReaders.addAll(newlyReservedReaders.keySet());
       return newlyReservedReaders;
     }
 
     public synchronized List<InterruptibleIterator> openFiles(Map<FileRef,DataFileValue> files, boolean detachable, SamplerConfigurationImpl samplerConfig)
         throws IOException {
 
-      List<FileSKVIterator> newlyReservedReaders = openFileRefs(files.keySet());
```

Comment:
This was the first time the lock is accessed.  Note even though this method is synchronized, its a different lock than openFiles.  The openFiles method is a tablet server wide lock.  This lock on this method is per scan.